src/index.js

```diff
@@ -1 +1,32 @@
-export default () => true
+import defaultsDeep from 'lodash.defaultsdeep';
+import pickBy from 'lodash.pickby';
+
+class Stipulate {
+
+  constructor(options) {
+    this.baseOptions = options;
```

Comment:
After talking with @morrisallison about this, it's a definite possibility for the future but we're blocking other projects, and we need to get a prototype out the door. I'll open a feature request for it which y'all can chat on.